### PR TITLE
[OpenCage] Update StreetName Locality and SubLocality Mapping

### PR DESCRIPTION
**Pull request by IvanBinzz**
Updated mapping according to official configuration https://github.com/OpenCageData/address-formatting/blob/master/conf/components.yaml

### CODE REVIEW

**Comment by jbelien** [comment]
Hello @IvanBinzz ,
Thanks a lot for your contributoin 👍 

Could you make sure your PR passes the tests ?
Thanks !

File: src/Provider/OpenCage/OpenCage.php [hunk 1]
```diff
@@ -248,7 +248,7 @@ protected function guessLocality(array $components)
      */
     protected function guessStreetName(array $components)
     {
-        $streetNameKeys = ['road', 'street', 'street_name', 'residential'];
+        $streetNameKeys = ['footway', 'street', 'street_name', 'residential', 'path', 'pedestrian', 'road_reference', 'road_reference_intl'];
```
Comment (eugenekurasov) [inline src/Provider/OpenCage/OpenCage.php]: road - is missing

Comment (jbelien) [reply src/Provider/OpenCage/OpenCage.php]: Indeed, thanks @eugenekurasov 👍

**Comment by eugenekurasov** [comment]
https://github.com/geocoder-php/Geocoder/blob/fb3ba5acb9182cbd3be4ac5fc6a3557ae3e4fa51/src/Provider/OpenCage/Tests/OpenCageTest.php#L110
Change to $this->assertEquals('Lancaster Gate', $result->getStreetName());

**Comment by jbelien** [comment]
Great work @IvanBinzz @eugenekurasov 🎉 

I'll merge this ASAP !

**Comment by jbelien** [comment]
I just released version `4.2.0`! 🎉

**Pull request merged**